File: packages/ios/SavePal/SavePal/Shared/Views/GoogleLogo.swift
import SwiftUI

struct GoogleLogo: View {
    var body: some View {
        Canvas { context, size in
            let w = size.width
            let h = size.height
            let cx = w / 2
            let cy = h / 2
            let r = min(w, h) / 2
            let innerR = r * 0.55
            let barH = r * 0.38

            // Blue (right arc: -45° to 45°)
            var blue = Path()
            blue.move(to: CGPoint(x: cx, y: cy))
            blue.addArc(center: CGPoint(x: cx, y: cy), radius: r,
                        startAngle: .degrees(-45), endAngle: .degrees(45), clockwise: false)
            blue.closeSubpath()
            context.fill(blue, with: .color(Color(red: 0.26, green: 0.52, blue: 0.96)))

            // Green (bottom-right arc: 45° to 135°)
            var green = Path()
            green.move(to: CGPoint(x: cx, y: cy))
            green.addArc(center: CGPoint(x: cx, y: cy), radius: r,
                         startAngle: .degrees(45), endAngle: .degrees(135), clockwise: false)
            green.closeSubpath()
            context.fill(green, with: .color(Color(red: 0.20, green: 0.66, blue: 0.33)))

            // Yellow (bottom-left arc: 135° to 225°)
            var yellow = Path()
            yellow.move(to: CGPoint(x: cx, y: cy))
            yellow.addArc(center: CGPoint(x: cx, y: cy), radius: r,
                          startAngle: .degrees(135), endAngle: .degrees(225), clockwise: false)
            yellow.closeSubpath()
            context.fill(yellow, with: .color(Color(red: 0.98, green: 0.74, blue: 0.02)))

            // Red (top-left arc: 225° to 315°)
            var red = Path()
            red.move(to: CGPoint(x: cx, y: cy))
            red.addArc(center: CGPoint(x: cx, y: cy), radius: r,
                       startAngle: .degrees(225), endAngle: .degrees(315), clockwise: false)
            red.closeSubpath()
            context.fill(red, with: .color(Color(red: 0.92, green: 0.26, blue: 0.21)))

            // White center circle
            var inner = Path()
            inner.addAllipse(in: CGRect(x: cx - innerR, y: cy - innerR, width: innerR * 2, height: innerR * 2))
            context.fill(inner, with: .color(.white))

            // Horizontal bar (the Google "G" crossbar) - blue color
            let barRect = CGRect(x: cx, y: cy - barH / 2, width: r * 0.95, height: barH)
            context.fill(Path(barRect), with: .color(Color(red: 0.26, green: 0.52, blue: 0.96)))
        }
    }
}

// Path extension since Canvas doesn't have addEllipse directly on Path via addAllipse typo guard
private extension Path {
    mutating func addAllipse(in rect: CGRect) {
        addEllipse(in: rect)
    }
}

#Preview {
    GoogleLogo()
        .frame(width: 40, height: 40)
}

File: packages/android/app/src/main/java/com/savepal/app/ui/components/GoogleLogo.kt
package com.savepal.app.ui.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Rect
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.drawscope.DrawScope
import androidx.compose.ui.unit.dp

private val GoogleBlue = Color(0xFF4285F4)
private val GoogleGreen = Color(0xFF34A853)
private val GoogleYellow = Color(0xFFFBBC05)
private val GoogleRed = Color(0xFFEA4335)

@Composable
fun GoogleLogo(modifier: Modifier = Modifier) {
    Canvas(modifier = modifier.size(18.dp)) {
        val w = size.width
        val h = size.height
        val cx = w / 2f
        val cy = h / 2f
        val r = minOf(w, h) / 2f
        val innerR = r * 0.55f
        val barH = r * 0.38f

        // Blue (right arc: -45° to 45°)
        drawArcWedge(cx, cy, r, -45f, 90f, GoogleBlue)

        // Green (bottom-right arc: 45° to 135°)
        drawArcWedge(cx, cy, r, 45f, 90f, GoogleGreen)

        // Yellow (bottom-left arc: 135° to 225°)
        drawArcWedge(cx, cy, r, 135f, 90f, GoogleYellow)

        // Red (top-left arc: 225° to 315°)
        drawArcWedge(cx, cy, r, 225f, 90f, GoogleRed)

        // White center circle
        drawCircle(Color.White, radius = innerR, center = Offset(cx, cy))

        // Horizontal bar (the Google "G" crossbar) - blue color
        drawRect(
            color = GoogleBlue,
            topLeft = Offset(cx, cy - barH / 2f),
            size = Size(r * 0.95f, barH)
        )
    }
}

private fun DrawScope.drawArcWedge(
    cx: Float, cy: Float, r: Float,
    startAngle: Float, sweepAngle: Float, color: Color
) {
    val path = Path().apply {
        moveTo(cx, cy)
        arcTo(
            rect = Rect(cx - r, cy - r, cx + r, cy + r),
            startAngleDegrees = startAngle,
            sweepAngleDegrees = sweepAngle,
            forceMoveTo = false
        )
        close()
    }
    drawPath(path, color)
}
